src/operations.jl

```diff
@@ -4,25 +4,75 @@ hash(c::TransparentGray, hx::UInt) = hash(alpha(c), hash(gray(c), hx))
 hash(c::AbstractRGB, hx::UInt) = hash(blue(c), hash(green(c), hash(red(c), hx)))
 hash(c::TransparentRGB, hx::UInt) = hash(alpha(c), hash(blue(c), hash(green(c), hash(red(c), hx))))
 
+# range
+range{T<:HSV}(::Type{T}) = (360,1,1)
+range{T<:HSL}(::Type{T}) = (360,1,1)
+range{T<:Lab}(::Type{T}) = (100,1,1)
+range{T<:LCHab}(::Type{T}) = (100,1,360)
+typealias Colorant1{T} Colorant{T,1}
+typealias Colorant2{T} Colorant{T,2}
+typealias Colorant3{T} Colorant{T,3}
+typealias Colorant4{T} Colorant{T,4}
+range{T<:Colorant1}(::Type{T}) = (1,)
+range{T<:Colorant2}(::Type{T}) = (1,1)
+range{T<:Colorant3}(::Type{T}) = (1,1,1)
+range{T<:Colorant4}(::Type{T}) = (1,1,1,1)
+
 # rand
-typealias RandTypes{T,C} Union{AbstractRGB{T},
-                               TransparentRGB{C,T},
-                               AbstractGray{T},
-                               TransparentGray{C,T}}
-typealias RandTypesFloat{C,T<:AbstractFloat} RandTypes{T,C}
-
-rand{G<:AbstractGray}(::Type{G}) = G(rand())
-rand{G<:TransparentGray}(::Type{G}) = G(rand(), rand())
-rand{C<:AbstractRGB}(::Type{C}) = C(rand(), rand(), rand())
-rand{C<:TransparentRGB}(::Type{C}) = C(rand(), rand(), rand(), rand())
-function rand{C<:RandTypesFloat}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(eltype(C), (sizeof(C)÷sizeof(eltype(C)), sz...)), sz)
+for t in [Float16,Float32,Float64,N0f8,N0f16,N0f32]
+    @eval _rand{T<:Union{AbstractRGB{$t},AbstractGray{$t}}}(::Type{T}) =
+          mapc(x->rand(eltype(T)), base_colorant_type(T)())
+end
+
+function _rand{T<:Colorant}(::Type{T})
+    R = range(T)
+    Mi = eltype(T) <: FixedPoint ? 1.0/typemax(eltype(T)) : 1.0
+    A = rand(eltype(T), length(T))
```

Comment:
Wouldn't `mapc((x,r)->r*Mi*rand(eltype(T)), base_colorant_type(T)(), R)` be considerably faster?